doc/rules/operator/new_with_braces.rst

```diff
@@ -4,29 +4,90 @@ Rule ``new_with_braces``
 
 All instances created with new keyword must be followed by braces.
 
+Configuration
+-------------
+
+``named_class``
+~~~~~~~~~~~~~~~
+
+Whether named classes should be followed by braces.
+
+Allowed types: ``bool``
+
+Default value: ``true``
+
+``anonymous_class``
```

Comment:
what's the purpose of having separated options for this? why using parentheses for named class, but not anonymous?

Reply:
They are different things, and I like having parentheses after normal classes, but not after anonymous ones (maybe because of misinterpreting PSR-12 and getting used to it). In both cases, I believe there are no rational reasons to use parentheses or not.

> The parentheses stress that a constructor is called and even needed when one or more arguments need to be passed.

You may as well say that parentheses are redundant because `new` keyword already indicates that the constructor is called. And, by the way, parentheses are added even when a class has no constructor.

One advantage of using parentheses even when no arguments are passed is that it's easier to add them later if needed, but that happens rarely with anonymous classes.